app/org/sagebionetworks/bridge/services/SurveyService.java

```diff
@@ -171,9 +183,22 @@ public void deleteSurveyPermanently(StudyIdentifier studyId, GuidCreatedOnVersio
 
         checkConstraintsBeforePhysicalDelete(studyId, keys);
 
+        // verify shared module existence as well
+        verifySharedModuleExistence(keys);
```

Comment:
Minor: For organizational purposes, it might be better to move this to checkConstraintsBeforePhysicalDelete()